File: LeetCode/Easy/JavaScript/0007reverseInteger.js
// LeetCode #7: Reverse Integer
// Difficulty: Easy
// Given a signed 32-bit integer x, return x with its digits reversed. If reversing x causes the value to go outside the signed 32-bit integer range [-231, 231 - 1], then return 0.

// Assume the environment does not allow you to store 64-bit integers (signed or unsigned).

var reverse = function(x) {
    let limit = 2147483647
    let negHold = x < 0 ? -1 : 1
    let str = String(Math.abs(x))
    let newStr = ''
    for (let i = str.length-1; i >= 0; i--) {
        newStr += str[i]
    }
    let revNum = Number(newStr)
    return revNum > limit ? 0 : revNum * negHold
};

File: LeetCode/Easy/Python3/0007reverse_integer.py
# LeetCode 0007: Reverse Integer
# Difficulty: Easy
# Given a signed 32-bit integer x, return x with its digits reversed. If reversing x causes the value to go outside the signed 32-bit integer range [-231, 231 - 1], then return 0.

# Assume the environment does not allow you to store 64-bit integers (signed or unsigned).


class Solution:
    def reverse(self, x: int) -> int:
        strNum = str(x)
        if x >= 0:
            revstr = strNum[::-1]
        else: 
            temp = strNum[1:]
            temp2 = temp[::-1]
            revstr = '-' + temp2
        if int(revstr) > 2**31-1 or int(revstr) <= -(2**31):
            return 0
        else:
            return int(revstr)

